src/python/py/models/builder.py

```diff
@@ -174,6 +178,10 @@ def make_genai_config(self, model_name_or_path, extra_kwargs, out_dir):
             },
         }
 
+        if self.ep == "cuda":
+            cuda_options = { "cuda" : { } }
```

Comment:
Could we generalize this and remove the `if` statement?

```suggestion
            ep_options = { self.ep : { } }
```

Reply:
Up to you, but won't any options inside here be provider specific?